src/DynamoCoreWpf/Views/FileTrust/FileTrustWarning.xaml.cs

```diff
@@ -134,10 +139,10 @@ private void YesButton_Click(object sender, RoutedEventArgs e)
             {
                 if (dynViewModel.PreferenceSettings.AddTrustedLocation(fileTrustWarningViewModel.DynFileDirectoryName))
                     dynViewModel.MainGuideManager.CreateRealTimeInfoWindow(string.Format(Properties.Resources.TrustLocationAddedNotification, fileTrustWarningViewModel.DynFileDirectoryName));
-            }
-            if (dynViewModel.CurrentSpaceViewModel.RunSettingsViewModel.Model.RunType != RunType.Manual)
+            }            
+            if ((dynViewModel.HomeSpaceViewModel as HomeWorkspaceViewModel).RunSettingsViewModel.Model.RunType != RunType.Manual)
             {
-                dynViewModel.Model.CurrentWorkspace.RequestRun();
+                (dynViewModel.HomeSpaceViewModel as HomeWorkspaceViewModel).Model.RequestRun();
```

Comment:
If we hide this dialog correctly, we should not run into case where `Yes` button is clicked on non-HomeSpaceViewModel. That was why I was asking you to hide it since it should not display in the first place. I dont think we need this check here..

Reply:
Done